### PR TITLE
RE2022-268: Add default parameters to match record

### PR DESCRIPTION
**Pull request by MrCreosote**
If the user omitted to include a default match parameter, it would not be included in the match record, but the default value would be used to process the match. This is especially problematic when saving provenance a set from a selection that was created from a match.

Closes https://github.com/kbase/collections/issues/316

### CODE REVIEW

File: src/service/matchers/lineage_matcher.py [hunk 1]
```diff
@@ -102,8 +103,7 @@ def generate_match_process(
         token - the user's token.
```
Comment (Xiangs18) [inline src/service/matchers/lineage_matcher.py:103]: token param is not used in this fun?

Comment (MrCreosote) [reply src/service/matchers/lineage_matcher.py:103]: Not for this matcher, but the minhash matcher (which implements the same method) does need it

File: src/service/processing_matches.py [hunk 2]
```diff
@@ -81,6 +83,19 @@ async def create_match(
     return curr_match
 
 
+def _process_user_params(params: dict[str, Any], matcher: Matcher):
+    ret = jsonschema_default.create_from(matcher.user_parameters)
+    if params:
+        ret = mergedeep.merge(ret, params, strategy=mergedeep.Strategy.REPLACE)
```
Comment (Xiangs18) [inline src/service/processing_matches.py:89]: Just wanna clarify here: 
When `ret` and `params` keys are the same, we take the value from `params`?

Comment (Xiangs18) [reply src/service/processing_matches.py:89]: That does not sound right ...

Comment (MrCreosote) [reply src/service/processing_matches.py:89]: Right. 2nd argument keys replace 1st

Comment (Xiangs18) [reply src/service/processing_matches.py:89]: that does not sound right ...

Comment (MrCreosote) [reply src/service/processing_matches.py:89]: Why not?

**Review by Xiangs18 — APPROVED** [review]
LGTM!

**Pull request merged**